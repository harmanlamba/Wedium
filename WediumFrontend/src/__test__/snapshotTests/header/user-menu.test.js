import React from 'react';
import { shallow } from 'enzyme';

import UserMenu from '../../../components/header/user-menu';

it('snapshot usermenu default load', () => {
    const user = {
        isAuthenticated: true,
        pictureUri: "https://imsteve.jpg",
        username: "steve mcqueen"
    }
    const setRouteLeaveHook = jest.fn();

    let wrapper = shallow(<
        UserMenu.WrappedComponent 
        user={user}
        params={
            {
                router: setRouteLeaveHook,
            }
        }
    />
    );

    expect(wrapper).toMatchSnapshot();
});