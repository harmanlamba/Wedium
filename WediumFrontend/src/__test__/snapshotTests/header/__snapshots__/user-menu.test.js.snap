// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot usermenu default load 1`] = `
<div>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-rightSpacing-4"
    color="primary"
    onClick={[Function]}
    startIcon={<UNDEFINED />}
    variant="contained"
  >
    Post
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-rightSpacing-4"
      display="inline"
      variant="button"
    >
      steve mcqueen
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-profileImage-5"
      src="https://imsteve.jpg"
    />
    <ExpandMoreIcon
      className="makeStyles-expand-1"
    />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    className="makeStyles-menuDropdown-3"
    getContentAnchorEl={null}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
  >
    <div>
      <WithStyles(ForwardRef(MenuItem))
        onClick={[Function]}
      >
        <PersonIcon
          className="makeStyles-rightSpacing-4"
        />
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          My Profile
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        onClick={[Function]}
      >
        <MeetingRoomIcon
          className="makeStyles-rightSpacing-4"
        />
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          Sign Out
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(MenuItem))>
    </div>
  </WithStyles(ForwardRef(Menu))>
</div>
`;
